### DISCUSSION TITLE
micro stutters (happens on every rendering setting)

### DISCUSSION THREAD

**Discussion by scarletremi** [ppy/osu#30515] [Q&A]
### Type

Performance

### Bug description

CPU: AMD Ryzen 5 3350G 
GPU: Radeon Vega Graphics
RAM: 8GB 

i keep getting micro-stutter regardless of any rendering option 

i've tried turning off and on hardware acceleration and tried switching to single thread, hardware acceleration off and single thread only fixed my input lag issues 

tried setting fps limiter on every setting but it still stutters 

### Version

2024.1009.1

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17635369/compressed-logs.zip)

**Comment by bdach** [comment]
Please add video evidence of this, preferably with frame graph shown (ctrl-f11 twice).

**Comment by Jerrynicki** [comment]
@bdach Hey, I'm a bit late to the party but I have the same issue and I come with video evidence :P
This happens to me on both of my laptops.

## Hardware/software specs
The first laptop (which I recorded the video on):
CPU: Ryzen AI 7 350
GPU: Radeon 860M iGPU
RAM: 32 GB

The second laptop:
CPU: Ryzen 5 5500U
GPU: integrated
RAM: 8 GB of RAM

Game version: 2026.518.0

Both are running Fedora Linux with GNOME on Wayland. 

The issue does _not_ happen on my PC, which has a dedicated GPU and is also running Linux + Wayland, but on openSUSE + Plasma.

## Things I've tried
Same as Scarletremi, changing to:
* Vulkan
* trying different settings in the FPS limiter
* disabling hardware acceleration
* disabling multithreaded rendering
did not change this issue occuring.

I also tried changing osu!'s process priority, with no results.

## Video and Logs
[Here's a video of it happening](https://youtu.be/p_qXUtNDMP8) (the first stutter is at ~1:00)

From the performance graph, it's visible that every ~10-20 seconds, there is a spike from "BufferSwap" to 20 ms, which causes noticeable micro-stutters, causing me to miss quite often when playing on my laptop, which I don't because of this issue :/

Two notes regarding the video:
* I'm running the game in Borderless because OBS wouldn't record it otherwise (I love wayland) - the issue happens the same way in Fullscreen, which is equivalent to Borderless on Wayland anyway
* The spikes from "Work" don't seem to happen when not recording, so I'm assuming they are caused by OBS encoding in the background

[1779366474.runtime.log](https://github.com/user-attachments/files/28100971/1779366474.runtime.log)
[1779366474.performance.log](https://github.com/user-attachments/files/28100979/1779366474.performance.log)
